aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -94,6 +106,53 @@ public void deleteFile(String path) {
     client().deleteObject(deleteRequest);
   }
 
+  /**
+   * Performs a batch deletion of the given paths. This implementation does not require
+   * that the passed in paths refer to objects which are in the same bucket.
+   * @param paths poths to delete
```

Comment:
nit: I would probably worded a bit differently like `Performs a batch deletion of given paths grouped by the bucket. Deletion will trigger whenever configured batch size reached for given bucket or for each bucket at the end.

also probably typo in param comment? `s/poths/paths`

Reply:
Good catch, and yeah I think the comment you have is more clear.